File: LeetCode/Java/Solution/src/ios/ac/cn/ConvertSortedArrayToBinarySearchTree.java
package ios.ac.cn;

public class ConvertSortedArrayToBinarySearchTree {
	private TreeNode consBST(int[] num, int start, int end) {
		if (start > end) {
			return null;
		}

		int mid = start + ((end - start) >> 1);	
		TreeNode root = new TreeNode(num[mid]);
		root.left = consBST(num, start, mid - 1);
		root.right = consBST(num, mid + 1, end);
		return root;
	}

	public TreeNode sortedArrayToBST(int[] num) {
		if (num == null || num.length == 0) {
			return null;
		}
		return consBST(num, 0, num.length - 1);
	}
	
	public static void main(String[] args) {
		ConvertSortedArrayToBinarySearchTree cbst = new ConvertSortedArrayToBinarySearchTree();
		int[] num = {1, 3, 5};
		System.out.println(cbst.sortedArrayToBST(num).val);
	}
}

File: LeetCode/Python/convert_sorted_array_to_binary_search_tree.py
#!/usr/bin/env python
#coding: utf-8

## Definition for a  binary tree node
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    # @param num, a list of integers
    # @return a tree node
    def sortedArrayToBST(self, num):
        if not num: return None
        ln = len(num)
        if ln == 1: return TreeNode(num[0])

        m = ln / 2
        root = TreeNode(num[m])
        root.left = self.sortedArrayToBST(num[:m])
        root.right = self.sortedArrayToBST(num[m+1:])
        return root
